src/wp-includes/taxonomy.php

```diff
@@ -684,7 +684,7 @@ function register_taxonomy_for_object_type( $taxonomy, $object_type ) {
 		return false;
 	}
 
-	if ( ! get_post_type_object( $object_type ) ) {
+	if ( 'user' !== $object_type && ! get_post_type_object( $object_type ) ) {
```

Comment:
`unregister_taxonomy_for_object_type` seems to need the same change (this fixes two failing tests).